cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -555,6 +556,94 @@ public void programSpecification(HttpRequest request, HttpResponder responder,
   }
 
   /**
+   * Update the log level for a program according to the request body. Currently supported program types are
+   * {@link ProgramType#FLOW}, {@link ProgramType#SERVICE} and {@link ProgramType#WORKER}.
+   * The request body is expected to contain a map of log levels, where key is loggername, value is one of the
+   * valid {@link org.apache.twill.api.logging.LogEntry.Level} or null.
+   */
+  @Path("/apps/{app-name}/{program-type}/{program-name}/logLevels")
+  @PUT
+  public void updateProgramLogLevels(HttpRequest request, HttpResponder responder,
+                                     @PathParam("namespace-id") String namespace,
+                                     @PathParam("app-name") String appName,
+                                     @PathParam("program-type") String type,
+                                     @PathParam("program-name") String programName) throws Exception {
+    updateProgramLogLevels(request, responder, namespace, appName, ApplicationId.DEFAULT_VERSION, type, programName);
+  }
+
+  /**
+   * Update the log level for a program according to the request body.
+   */
+  @Path("/apps/{app-name}/versions/{app-version}/{program-type}/{program-name}/logLevels")
+  @PUT
+  public void updateProgramLogLevels(HttpRequest request, HttpResponder responder,
```

Comment:
This is how it is supposed to work, we use REST endpoint to change the log level of programs at runtime. And we already have #6705  which allows you to use preferences to set log levels per program run. How can we make a it clearer for the user?